tests/test_ote_training.py

```diff
@@ -0,0 +1,617 @@
+        task_cls = get_task_class(task_impl_path)
+        task = task_cls(task_environment=environment)
+        return environment, task
+
+    def _run_ote_training(self):
+        logger.debug(f'self.template_file_path = {self.template_file_path}')
+        logger.debug(f'Using for train annotation file {self.dataset_params.annotations_train}')
+        logger.debug(f'Using for val annotation file {self.dataset_params.annotations_val}')
+
+        self.dataset = MMDatasetAdapter(
+            train_ann_file=self.dataset_params.annotations_train,
+            train_data_root=self.dataset_params.images_train_dir,
+            val_ann_file=self.dataset_params.annotations_val,
+            val_data_root=self.dataset_params.images_val_dir,
+            test_ann_file=self.dataset_params.annotations_test,
+            test_data_root=self.dataset_params.images_test_dir,
+            dataset_storage=NullDatasetStorage)
+
+        self.labels_schema = generate_label_schema(self.dataset.get_labels())
+        labels_list = self.labels_schema.get_labels(False)
+        self.dataset.set_project_labels(labels_list)
+        print(f'train dataset: {len(self.dataset.get_subset(Subset.TRAINING))} items')
+        print(f'validation dataset: {len(self.dataset.get_subset(Subset.VALIDATION))} items')
+
+        logger.debug('Load model template')
+        self.model_template = parse_model_template(self.template_file_path)
+
+        hyper_parameters = self.model_template.hyper_parameters.data
+        set_values_as_default(hyper_parameters)
+
+        logger.debug('Setup environment')
+        params = create(hyper_parameters)
+        logger.debug('Set hyperparameters')
+        params.learning_parameters.num_iters = self.num_training_iters
+        if self.num_training_iters < 20:
+            num_checkpoints = 2
+        elif self.num_training_iters < 1000:
+            num_checkpoints = 10
+        else:
+            num_checkpoints = 30
+
+        #### TODO: fixed parameter, delete this
+        params.learning_parameters.batch_size = 2
+
+        params.learning_parameters.num_checkpoints = num_checkpoints
+
+        self.environment, self.task = self._create_environment_and_task(params,
+                                                                        self.labels_schema,
+                                                                        self.model_template)
+
+        logger.debug('Train model')
+        self.output_model = ModelEntity(
+            self.dataset,
+            self.environment.get_model_configuration(),
+            model_status=ModelStatus.NOT_READY)
+
+        self.copy_hyperparams = deepcopy(self.task._hyperparams)
+
+        self.task.train(self.dataset, self.output_model)
+        logger.info(f'performance={self.output_model.performance}')
+
+        # To check that the training performance is valid
+        # and may be converted to score name and value
+        self.get_training_performance_as_score_name_value()
```

Comment:
Please note, that you access the attribute first for printing its value, and only then check that attribute exists.

Reply:
Fixed